### PR TITLE
feat(calimero-sdk + node): update types and logic for new context updates

### PR DESCRIPTION
**Pull request by frdomovic**
# feat(calimero-sdk + node): update types and logic for new context updates

## Summary

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (Add Jira tickets related to this change)

## Test plan

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.
Please also list any relevant details for your test configuration. Add screenshots or videos for changes in the user-interface.

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have added a test plan that prove my fix is effective or that my feature works
- [ ] I squashed all commits and provided a meaningful commit message
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: crates/server/src/admin/storage/client_keys.rs [hunk 1]
```diff
@@ -32,8 +32,9 @@ pub fn get_context_client_key(
     let did = get_or_create_did(store)?;
     Ok(did
         .client_keys
-        .into_iter()
-        .filter(|k| &k.context_id == context_id)
+        .iter()
+        .filter(|k| k.context_id.as_ref() == Some(context_id))
+        .cloned()
```
Comment (miraclx) [inline crates/server/src/admin/storage/client_keys.rs]:
```suggestion
        .into_iter()
        .filter(|k| k.context_id == Some(context_id))
```

this doesn't work?

File: packages/calimero-sdk/src/subscriptions/subscriptions.ts [hunk 2]
```diff
@@ -15,7 +15,7 @@ export interface SubscriptionsClient {
 export type NodeEvent = ApplicationEvent;
 
 export interface ApplicationEvent {
-  application_id: ApplicationId;
```
Comment (miraclx) [inline packages/calimero-sdk/src/subscriptions/subscriptions.ts:18]: weird this wasn't already `camelCase`

seems right - https://github.com/calimero-network/core/blob/83c09bb45dc46d6cd119471a960e70a1e93867a1/crates/primitives/src/events.rs#L13-L15

**Review by miraclx — APPROVED** [review]

File: crates/server/src/admin/storage/client_keys.rs (hunk 1 above)
Comment (frdomovic) [reply crates/server/src/admin/storage/client_keys.rs]: 1.
```
can't compare `std::option::Option<ContextId>` with `std::option::Option<&ContextId>`
the trait `PartialEq<std::option::Option<&ContextId>>` is not implemented for `std::option::Option<ContextId>`
the following other types implement trait `PartialEq<Rhs>`:
  <std::option::Option<Box<U>> as PartialEq<rkyv::niche::option_box::ArchivedOptionBox<T>>>
  <std::option::Option<T> as PartialEq>
  <std::option::Option<U> as PartialEq<rkyv::option::ArchivedOption<T>>>rustc
```
2.
```
a value of type `Vec<ClientKey>` cannot be built from an iterator over elements of type `&ClientKey`
the trait `FromIterator<&ClientKey>` is not implemented for `Vec<ClientKey>`
the trait `FromIterator<ClientKey>` is implemented for `Vec<ClientKey>`
for that trait implementation, expected `ClientKey`, found `&ClientKey`rustc[Click for full compiler diagnostic](rust-analyzer-diagnostics-view:/diagnostic%20message%20%5B1%5D?1#file%3A%2F%2F%2FUsers%2Ffrandomovic%2FDesktop%2Fcali2.0-experimental%2Fcrates%2Fserver%2Fsrc%2Fadmin%2Fstorage%2Fclient_keys.rs)
client_keys.rs(35, 10): the method call chain might not have had the expected associated types
iterator.rs(2000, 19): required by a bound in `std::iter::Iterator::collect`
```

Comment (miraclx) [reply crates/server/src/admin/storage/client_keys.rs]: in that case, restoring `as_ref` but keeping it `into_iter` should work

Comment (frdomovic) [reply crates/server/src/admin/storage/client_keys.rs]:
```
a value of type `Vec<ClientKey>` cannot be built from an iterator over elements of type `&ClientKey`
the trait `FromIterator<&ClientKey>` is not implemented for `Vec<ClientKey>`
the trait `FromIterator<ClientKey>` is implemented for `Vec<ClientKey>`
for that trait implementation, expected `ClientKey`, found `&ClientKey`rustc[Click for full compiler diagnostic](rust-analyzer-diagnostics-view:/diagnostic%20message%20%5B0%5D?0#file%3A%2F%2F%2FUsers%2Ffrandomovic%2FDesktop%2Fcali2.0-experimental%2Fcrates%2Fserver%2Fsrc%2Fadmin%2Fstorage%2Fclient_keys.rs)
client_keys.rs(35, 10): the method call chain might not have had the expected associated types
iterator.rs(2000, 19): required by a bound in `std::iter::Iterator::collect`
```

Comment (miraclx) [reply crates/server/src/admin/storage/client_keys.rs]: [`f391cff` (#437)](https://github.com/calimero-network/core/pull/437/commits/f391cffa3fc5ec963d7955af02ff8f27f17d25a8)

Comment (frdomovic) [reply crates/server/src/admin/storage/client_keys.rs]: thanks

File: packages/calimero-sdk/src/subscriptions/subscriptions.ts [hunk 3]
```diff
@@ -15,7 +15,7 @@ export interface SubscriptionsClient {
...
+  context_id: ContextId;
```
Comment (miraclx) [inline packages/calimero-sdk/src/subscriptions/subscriptions.ts:18]: can we fix this side now?

Comment (frdomovic) [reply packages/calimero-sdk/src/subscriptions/subscriptions.ts:18]: sure

Comment (frdomovic) [reply packages/calimero-sdk/src/subscriptions/subscriptions.ts:18]: Should i just add rename camelCase ?

Comment (miraclx) [reply packages/calimero-sdk/src/subscriptions/subscriptions.ts:18]: it's already this on the rust side, I believe

https://github.com/calimero-network/core/blob/134d88f28d9f99b7bc5a3ee68122918553b9c4e9/crates/primitives/src/events.rs#L12-L15

**Pull request merged**